### PR TITLE
Sockets9361

### PR DESCRIPTION
**Pull request by ranechita**
Added support for linux sockets for ad9361. The line with ad9361 dev is commented at the moment

### CODE REVIEW

File: network/linux_socket/linux_socket.c [hunk 1]
```diff
@@ -212,14 +212,14 @@ static int32_t linux_socket_bind(struct linux_desc *desc, uint32_t sock_id,
 	int32_t ret;
 	struct sockaddr_in saddr = {0};
 	socklen_t len;
+	int32_t option = 1;
```
Comment (mchindri) [inline network/linux_socket/linux_socket.c:215]: This change is not needed here

File: network/linux_socket/linux_socket.c [hunk 2]
```diff
@@ -105,7 +105,7 @@ static int32_t linux_socket_connect(struct linux_desc *desc, uint32_t sock_id,
 	hptr = gethostbyname(addr->addr);
 	saddr.sin_addr.s_addr = ((struct in_addr*) hptr->h_addr_list[0])->s_addr;
 	socklen_t len = sizeof(saddr);
-
+
```
Comment (mchindri) [inline network/linux_socket/linux_socket.c:108]: Change not needed

File: network/network_interface.h [hunk 3]
```diff
@@ -51,6 +51,8 @@
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
+#define MAX_BACKLOG 0xFFFFFFFF
```
Comment (mchindri) [inline network/network_interface.h:54]: Maybe is better to add this in tcp_socket.h

File: projects/ad9361/src.mk [hunk 4]
```diff
@@ -58,10 +79,11 @@ INCS +=	$(INCLUDE)/axi_io.h						\
 	$(INCLUDE)/delay.h						\
 	$(INCLUDE)/util.h
 ifeq (y,$(strip $(TINYIIOD)))
+INCS += $(NO-OS)/iio/iio_app/iio_app.h 
+
 INCS += $(INCLUDE)/fifo.h						\
 	$(INCLUDE)/uart.h						\
 	$(INCLUDE)/list.h						\
-	$(PLATFORM_DRIVERS)/uart_extra.h				\
```
Comment (mchindri) [inline projects/ad9361/src.mk:64]: This may brake some projects

File: projects/ad9361/src/main.c [hunk 5]
```diff
@@ -70,8 +73,12 @@
 #include "iio_axi_dac.h"
 #include "iio_ad9361.h"
 #include "uart.h"
+#include "iio_app.h"
```
Comment (mchindri) [inline projects/ad9361/src/main.c:76]: A better description will be:
Update project to use iio_app

File: projects/ad9361/src/main.c [hunk 6]
```diff
@@ -785,7 +793,7 @@ int main(void)
 	 */
 	struct xil_uart_init_param xil_uart_init_par;
 
-	/**
+    /**
```
Comment (mchindri) [inline projects/ad9361/src/main.c:796]: Extra change

File: projects/ad9361/src/main.c (hunk 5 above)
Comment (mchindri) [reply projects/ad9361/src/main.c:76]: And update xilinx code to use iio_app to :)

File: projects/ad9361/src/parameters.h [hunk 7]
```diff
@@ -148,6 +148,9 @@
 #define CF_AD9361_TX_DMA_BASEADDR	3
 
 #define SPI_DEVICE_ID		0
+#define DAC_DDR_BASEADDR	0x10000000
```
Comment (mchindri) [inline projects/ad9361/src/parameters.h:151]: This may be an issue. In linux you can't access raw memory address. 
Try using an approach like this: https://github.com/analogdevicesinc/no-OS/blob/b34afe2704b607c0789b91b25a0f875236616a0b/projects/iio_demo/src/app/main.c#L86

**Comment by ranechita** [comment]
v2

- removed unnecessary spaces and variables from linux_sockets
- modified linux sockets commit message

**Comment by ranechita** [comment]
v3

- moved MAX_BACKLOG from network_interface.h
- ran astyle

**Comment by ranechita** [comment]
v4

- modified ADC/DAC_DDR_BASEADDR

File: network/linux_socket/linux_socket.c [hunk 8]
```diff
@@ -105,7 +105,6 @@ static int32_t linux_socket_connect(struct linux_desc *desc, uint32_t sock_id,
 	hptr = gethostbyname(addr->addr);
```
Comment (adrimbarean) [inline network/linux_socket/linux_socket.c:105]: Could you rephrase the commit message? Seems a bit ambiguous to me.

File: network/linux_socket/linux_socket.c [hunk 9]
```diff
@@ -249,14 +247,14 @@ static int32_t linux_socket_accept(struct linux_desc *desc, uint32_t sock_id,
 
 	ret = accept(sock_id, NULL, NULL);
 
-	flags = fcntl(sock_id, F_GETFL);
```
Comment (adrimbarean) [inline network/linux_socket/linux_socket.c:252]: Maybe [accept4](https://man7.org/linux/man-pages/man2/accept.2.html) could also be used here with the `SOCK_NONBLOCK` flag to make it non-blocking as soon as accepted.

File: projects/ad9361/src/main.c [hunk 10]
```diff
@@ -70,14 +73,22 @@
...
 
 #endif // IIO_SUPPORT
 
 /******************************************************************************/
 /************************ Variables Definitions *******************************/
 /******************************************************************************/
+
+#ifdef LINUX_PLATFORM
+#endif
+
```
Comment (adrimbarean) [inline projects/ad9361/src/main.c:91]: Is there a purpose to this?

File: projects/ad9361/src/main.c [hunk 11]
```diff
@@ -877,14 +881,14 @@ int main(void)
...
-	if (status < 0)
-		return status;
 
-	do {
-		status = iio_step(iio_app_desc);
-	} while (true);
+	struct iio_app_device devices[] = {
+		IIO_APP_DEVICE("axi_adc", iio_axi_adc_desc, adc_dev_desc, &read_buff, NULL),
+		IIO_APP_DEVICE("axi_dac", iio_axi_dac_desc, dac_dev_desc, &write_buff, NULL),
+		//IIO_APP_DEVICE("ad9361", ad9361_phy, ad9361_dev_desc, NULL, NULL)
```
Comment (adrimbarean) [inline projects/ad9361/src/main.c:888]: Let's remove commented code if we don't need it.

File: projects/ad9361/src/main.c (hunk 10 above)
Comment (ranechita) [reply projects/ad9361/src/main.c:91]: sorry, i was trying something right now and i pushed that by mistake, i'll remove it asap

File: projects/ad9361/src/main.c (hunk 11 above)
Comment (ranechita) [reply projects/ad9361/src/main.c:888]: the commented line is the one that has been causing some issues and I left it there in case anyone wanted to test it

**Comment by ranechita** [comment]
v5

- remove useless ifdef
- add BAUDRATE in parameters.h, so that xlinix can also use iio
- updated src.mk for xilinx

File: projects/ad9361/src/main.c [hunk 12]
```diff
@@ -811,19 +818,20 @@ int main(void)
...
-		return status;
-
 	status = uart_init(&uart_desc, &uart_init_par);
 	if (status < 0)
 		return status;
 
+	status = axi_dmac_init(&tx_dmac, &tx_dmac_init);
+	if(status < 0)
+		return status;
+
```
Comment (mchindri) [inline projects/ad9361/src/main.c:828]: Why was this moved?

**Comment by mchindri** [comment]
Also, in an other commit you can change struct linux_desc *desc -> void *desc;
linux_desc doesn't exist and a warning is generated
